pkg/linter/linter.go

```diff
@@ -314,33 +314,24 @@ func (lctx LinterContext) LintPackageFs(fsys fs.FS, warn func(error), linters []
 				continue
 			}
 
-			err = linter.LinterFunc(lctx, path, d)
-			if err != nil {
-				if linter.FailOnError {
-					return fmt.Errorf("Linter %s failed at path %q: %w; suggest: %s", linterName, path, err, linter.Explain)
-				}
-				warn(err)
+			if err := linter.LinterFunc(lctx, path, d); err != nil {
+				errs = append(errs, err)
 			}
 		}
-
-		return nil
-	}
-
-	if err := fs.WalkDir(fsys, ".", walkCb); err != nil {
+		return errors.Join(errs...)
+	}); err != nil {
 		return err
 	}
 
 	// Run post-walking linters
+	var errs []error
 	for _, linterName := range postLinters {
 		linter := postLinterMap[linterName]
-		err := linter.LinterFunc(lctx, fsys)
-		if err != nil {
-			if linter.FailOnError {
-				return fmt.Errorf("Linter %s failed; suggest: %s", linterName, linter.Explain)
-			}
-			warn(err)
```

Comment:
This eliminates the ability to have individual linters be fatal.

Reply:
Basically: we need a way to introduce a linter as just a warning, get it clean, and then make it fatal.

If we land this then we can't make anything required without making everything required and it will become a requirement to get linters clean across Wolfi before landing any of its logic, which I don't think we want.

Reply:
WDYT about having `--fail-on-lint-warning=empty,setuidgid,foo,bar` which selects which ones will be considered fatal?

Reply:
That seems fine, but I still think linters should be able to have defaults.  This could be through the flag defaults, I have no preference here.